discovery/gossiper.go

```diff
@@ -1099,18 +1099,28 @@ func splitAnnouncementBatches(subBatchSize int,
 func (d *AuthenticatedGossiper) sendBatch(announcementBatch []msgWithSenders) {
 	syncerPeers := d.syncMgr.GossipSyncers()
 
-	// We'll first attempt to filter out this new message
-	// for all peers that have active gossip syncers
-	// active.
+	// announcedPeers is a map of peers that have successful sent messages.
+	// We will ignore these peers when broadcasting the announcementBatch.
+	announcedPeers := make(map[route.Vertex]*GossipSyncer, len(syncerPeers))
+
+	// We'll first attempt to filter out this new message for all peers
+	// that have active gossip syncers active.
 	for _, syncer := range syncerPeers {
-		syncer.FilterGossipMsgs(announcementBatch...)
+		announced := syncer.FilterGossipMsgs(announcementBatch...)
```

Comment:
So I started to do something like this, but then realized it's possible that:
  * They're able to process some messages (we try to batch anns here), but not our own locally sourced gossip message 

We'd then fail to send it to them below since we've tacked them as an announced peer.

Reply:
Yeah I think this PR doesn't fix the spec part, so #7239 needs to be merged.

> They're able to process some messages (we try to batch anns here), but not our own locally sourced gossip message

Becasue of the timestamp filter?

What I've noticed and locally reproduced is the case when the race happens between the events,
- the receiver creates and sends the `GossipTimestampRange`, and,
- meanwhile the sender has some messages, whether they are generated locally or received from the remote,

It's likely these messages would never be relayed because the timestamp filter used in syncers and the peers being skipped.

Reply:
Ah yeah, good point. I think the pinned syncers fix can get in though